browser-tests/snapshot-tests/snapshot.spec.js

```diff
@@ -0,0 +1,104 @@
+const { test, expect } = require("@playwright/test");
+const path = require("path");
+const fs = require("fs");
+
+test("Snapshots", async ({ page }) => {
+
+  test.setTimeout(120000)
+  for (const pageName of Object.keys(pagesToTest)) {
+    const contexts = pagesToTest[pageName];
+    const contextsToTest = contexts.length > 0 ? contexts : [ undefined ];
+
+    for (const context of contextsToTest) {
+      for (const language of [ "en", "cy" ]) {
+        var url = `http://localhost:4601/dev/template/${pageName}/${language}`;
+        var screenshotContext = "";
+        if (context !== undefined) {
+          url += `?context=${context}`;
+          screenshotContext = `-${context}`;
+        }
+        const screenshotFilename = `${pageName}-${language}${screenshotContext}.png`;
+        await page.goto(url);
+        await expect(page).toHaveScreenshot(screenshotFilename, { fullPage: true });
+      }
+    }
+  }
+});
+
+test("All templates have snapshot tests", async () => {
+  const directoryPath = path.join(__dirname, "/../../src/views/ipv/page");
+  const missingTemplates = [];
+
+  fs.readdir(directoryPath, function (err, files) {
+    if (err) {
+      throw err;
+    }
+
+    const testedTemplates = Object.keys(pagesToTest);
+
+    for (const file of files) {
+      const templateName = path.parse(file).name;
+      if (!testedTemplates.find(tt => tt === templateName)) {
+        missingTemplates.push(templateName);
+      }
+    }
+
+    expect(missingTemplates).toStrictEqual([]);
+  });
+});
+
+const pagesToTest = {
```

Comment:
Would it be possible to spread over all of the views defined, then you can explicitly redefine ones which want contexts?

```javascript
const pagesToTest = {
  ...Object.fromEntries(
    fs.readdirSync("/../../src/views/ipv/page")
      .map(file => [parse(file).name, []])
  ),
  "no-photo-id-security-questions-find-another-way": ["dropout", undefined],
  // ...
  "pyi-triage-desktop-download-app": ["android", "iphone"],
  "update-details-failed": ["repeatFraudCheck",undefined],
  "update-name-date-birth": ["repeatFraudCheck","reuse","rfcAccountDeletion",undefined]
}
```

Then I don't think we'd need the above test

Reply:
One of the points of the test is to make sure people think about adding pages to the snapshots - that way they won't forget to include contexts (which we have no way of automatically finding)